### ISSUE TITLE
Dependency on Autofac 4.6 not required

### ISSUE THREAD

**Issue by SamuelDebruyn** [linvi/tweetinvi#722]
My current project has a strict dependency on the maximum version of Autofac being 4.5. I'd like to use Tweetinvi, but I can't because it has a dependency on Autofac 4.6.

However, this doesn't seem to be a strict dependency, because the project builds fine with Autofac 4.5.

Is it possible to lower the Autofac minimum version dependency to 4.5?

**Comment by linvi** [comment]
I will investigate but I am not sure this is possible for all the platforms. Usually I check to ensure that all type of platforms are supported and work as expected.

**Comment by linvi** [comment]
In the meantime, I would suggest that you download Tweetinvi and compile it on your side.
The project is very easy to build ;)

**Comment by AmitKumar-AK** [comment]
@linvi I am trying to build the project using MS VS Enterprise 2017 and getting error related to The type 'Attribute' is defined in an assembly that is not referenced. You must add a reference to assembly 'netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'.. Also not able to change the netstandard version. I wanted to change tweetinvi, .net framework to 4.7.1.

**Comment by linvi** [comment]
Hello. I have tried reinstalling and compiling with VS 2017 and did not get any issue. 

Have you installed the netstandard/encore sdk?

**Comment by AmitKumar-AK** [comment]
No. I wanted to use it with .Net Framework 4.7.1 and existing Nuget package not compatible, so need to upgrade it.
Can you please share the link to install netstandard/encore sdk?
I changed the csproject file and able to build some solution but whole solution not compiling successfully

**Comment by linvi** [comment]
Tweetinvi is fully based on netstandard which is compatible with .NET framework. Go to Visual Studio installer and when modifying you should have options. 

I am not on my computer right now but this is fairly simple to find.

**Comment by AmitKumar-AK** [comment]
If I am trying to utilise Nuget package latest version on .net framework 4.7.1 based project then it's not allowing,due to lesser version of Tweetinvi.

You can check.

**Comment by AmitKumar-AK** [comment]
In VS 2017 it's giving me error to add reference of netstandard 2.0 assembly but not able to change version from manage Nuget package options because upgrade option disable for higher version.

**Comment by linvi** [comment]
Tweetinvi is currently building towards .NET Standard 1.4.
It is therefore compatible with .NET Framework 4.7.1.

If you are trying to build tweetinvi you just need to install the SDK...

https://docs.microsoft.com/en-us/dotnet/standard/net-standard

**Comment by AmitKumar-AK** [comment]
As per table compatibility with .net framework 4.7.1 not mentioned for .net standard.

**Comment by linvi** [comment]
.net standard 1.4 supports all versions of .net framework above 4.6.1.

> The following table lists the minimumplatform versions that support each .NET Standard version. That means that later versions of a listed platform also support the corresponding .NET Standard version. For example, .NET Core 2.2 supports .NET Standard 2.0 and earlier.

**Comment by AmitKumar-AK** [comment]
It's not, please try creating the 4.7.1 based project and use the Nuget package

**Comment by linvi** [comment]
In addition

> For .NET Framework projects that need to use such libraries, we recommend that you upgrade the project to target .NET Framework 4.7.2 or higher.

**Comment by linvi** [comment]
Tried without any problem on version 4.7.1.

**Comment by AmitKumar-AK** [comment]
With Nuget package?

BR,
Amit Kumar
Sent from Mobile. Please ignore typos & brevity.

________________________________

**Comment by AmitKumar-AK** [comment]
I converted the code for .net framework 4.7.1 by changing the csproject files but i am getting authenticatedUser = Id = 18, Status = WaitingForActivation, Method = "{null}", Result = "{Not yet computed}" response while var authenticatedUser = UserAsync.GetAuthenticatedUser(creds);

Any pointers to resolve this issue?

**Comment by linvi** [comment]
@AmitKumar-AK sorry but you need to understand how to use .NET for using this library. Please read more about .NET.

In the meantime for your question :

``` c#
// this is returning a Task<IAuthenticatedUser> so you need to await for the result
var authenticatedUser = await UserAsync.GetAuthenticatedUser(creds);

// or use non async  methods
var authenticatedUser = await User.GetAuthenticatedUser(creds);
```

**Comment by AmitKumar-AK** [comment]
I followed below mentioned code but still getting null for authenticated user in ASP.NET MVC Project:
		[ActionName("About")]
		public async Task<ActionResult> SearchAsyncTweetinvi()
		{
			ViewBag.Message = "Your application description page.";
			IAuthenticatedUser x;
			string CONSUMER_KEY = "";
			string CONSUMER_SECRET = "";
			string ACCESS_TOKEN = "";
			string ACCESS_TOKEN_SECRET = "";
			var creds = new TwitterCredentials(CONSUMER_KEY, CONSUMER_SECRET, ACCESS_TOKEN, ACCESS_TOKEN_SECRET);
			var authenticatedUser = await UserAsync.GetAuthenticatedUser(creds);
			x = authenticatedUser;


			return View();
		}

**Comment by linvi** [comment]
Hi. `null` means something went wrong. To receive more information you can disable the ExceptionHandler as explained here : https://github.com/linvi/tweetinvi/wiki/Exception-Handling#global-application-handling-mechanism.

Let me know what the error is and I will be able to help you further.

**Comment by AmitKumar-AK** [comment]
((System.Exception)latestException).Message = The underlying connection was closed: An unexpected error occurred on a send.

((Tweetinvi.Exceptions.TwitterException)latestException).TwitterDescription coming as null.

**Comment by AmitKumar-AK** [comment]
((System.Exception)latestException).Message = The underlying connection was closed: An unexpected error occurred on a send.

((Tweetinvi.Exceptions.TwitterException)latestException).TwitterDescription coming as null.
Qury url : https://api.twitter.com/1.1/account/verify_credentials.json?skip_status=true&include_entities=true&include_email=true

**Comment by linvi** [comment]
Hi,

Please check this one out : https://github.com/linvi/tweetinvi/wiki/ASP.NET-consideration

Cheers,
Linvi

**Comment by AmitKumar-AK** [comment]
ex = {"https://api.twitter.com/1.1/account/verify_credentials.json?skip_status=true&include_entities=true&include_email=true web request timed out."}

Twitter was not able to perform your query within the Timeout limit of 10000 ms.

**Comment by linvi** [comment]
Did this get solved?

**Comment by AmitKumar-AK** [comment]
Not yet. I am planning to spin up the Win Azure VM and will try inside that to check the n/w error. May be organisation firewall blocking the request to twitter api's.

**Comment by linvi** [comment]
Any luck here with the firewall?

**Comment by AmitKumar-AK** [comment]
Not tried yet in VM. Will update you.
